test/CMakeLists.txt

```diff
@@ -177,3 +184,25 @@ add_umf_test(NAME base_alloc_linear
 add_umf_test(NAME base_alloc_global
             SRCS ${BA_SOURCES_FOR_TEST} pools/pool_base_alloc.cpp malloc_compliance_tests.cpp
             LIBS umf_utils)
+
+add_umf_test(NAME pool_os_alloc
+            SRCS pools/pool_os_alloc.cpp malloc_compliance_tests.cpp)
+
+# tests for the proxy library
+if((LINUX OR WINDOWS) AND UMF_BUILD_OS_MEMORY_PROVIDER AND UMF_BUILD_SHARED_LIBRARY)
+    if(WINDOWS AND NOT (CMAKE_BUILD_TYPE STREQUAL Release))
+        message(STATUS "Skipping the tests of the proxy library, because they are supported only in the Release build on Windows")
```

Comment:
Because it does not work on Windows in case of the Debug build - there is the `_free_dbg()` function called instead of `free()` that cannot be found for unknown reason (debugged with @bratpiorka)

Reply:
Ok, I see, but that's a problem for the entire proxy pool, not only in tests right? Should have this check in the main CMake? Also, we should probably add some todo and an issue to track this.

Reply:
Right. Done: https://github.com/oneapi-src/unified-memory-framework/issues/350